### DISCUSSION TITLE
Version Policy discussion

### DISCUSSION THREAD

**Discussion by jijingg** [SpinalHDL/SpinalHDL#862] [General]
With the growing scale of Spinalhdl, more and more people are participating. Due to the testing problem, the current version strategy may not be suitable, because it is difficult for us to release a stable enough version.
So I suggest that when we release a large version, for example, 1.7.1 is a version with temporary problems,  should we release some small versions 1.7.1. rc1 and 1.7.1. rc2 first, and then release 1.7.1 after two weeks when we fix the bug from early adopters of new version
so that all our large versions are at least relatively stable.

ref: https://en.wikipedia.org/wiki/Software_release_life_cycle

**Comment by Dolu1990** [comment]
Hi,

So far, the idea was to release directly 1.7.1 version, and if there is a bad bug, to release some hot fix as 1.7.1a 1.7.1b 1.7.1c and so on. So sort of a light weight aproache, as we kind of never know when a given version is bug free and there is kind of no schedule.

So between :
- 1.7.1 1.7.1a 1.7.1b 1.7.1c
- 1.7.1 rc1 1.7.1 rc2 1.7.1 

I'm not against having the proposed version skim,  but would there be enough people testing stuff to make a difference ?

**Reply by jijingg** [reply]
> I'm not against having the proposed version skim, but would there be enough people testing stuff to make a difference ?
 
Even if there are not many people testing, if there are serious problems with the new version, as long as someone uses it, their previous code itself is a test set, and someone will always crash the old code or produce bad results.

1.7.1 1.7.1a 1.7.1b 1.7.1c  is ok， At least let people know which version is relatively stable.

**Reply by Dolu1990** [reply]
> 1.7.1 1.7.1a 1.7.1b 1.7.1c is ok， At least let people know which version is relatively stable.

Ok, good to me :D

**Comment by andreasWallner** [comment]
Not that I really have a stake in that, but it might make sense do adopt schemes that people (and tools like sbt (e.g. when requesting a version ">=1.7") are already used to like semver: https://semver.org/.
That way you would:
Given a version number MAJOR.MINOR.PATCH, increment the:

1. MAJOR version when you make incompatible API 
2. MINOR version when you add functionality in a backwards compatible manner
3. PATCH version when you make backwards compatible bug fixes

Which I felt was anyhow what you were already doing?

**Reply by Dolu1990** [reply]
Yes, it is what what done, just that in the PATCH level were also included "small" feature progresses / addition, and that hot fix were in PATCH followed by a character (a b c d ex  1.7.1b)

**Comment by Dolu1990** [comment]
One thing also, is that if somebody is enoyed by a bug in the last version released and the bug is fixed in dev, to notify me, so a hotfix can be released. I'm not necessarly aware ^^

**Comment by Dolu1990** [comment]
Just released one hot fix : https://github.com/SpinalHDL/SpinalHDL/releases/tag/v1.7.3a
